health/metrics/plugin_impl_metrics.go

```diff
@@ -0,0 +1,234 @@
+// Copyright (c) 2017 Cisco and/or its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package metrics implements Prometheus health/metrics handlers.
+package metrics
+
+import (
+	"net/http"
+
+	"github.com/ligato/cn-infra/core"
+	"github.com/ligato/cn-infra/flavors/local"
+	"github.com/ligato/cn-infra/health/statuscheck"
+	"github.com/ligato/cn-infra/logging"
+	log "github.com/ligato/cn-infra/logging/logroot"
+	"github.com/ligato/cn-infra/rpc/rest"
+	"github.com/ligato/cn-infra/utils/safeclose"
+	"github.com/namsral/flag"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/unrolled/render"
+)
+
+const (
+	defaultPluginName  string = "HEALTH-METRICS"
+	DefaultMetricsPath string = "/metrics" // default Prometheus metrics URL
+	agentName          string = "agent"
+	namespace          string = ""
+	subsystem          string = ""
+	serviceLabel       string = "service"
+	dependencyLabel    string = "dependency"
+	buildVersionLabel  string = "build_version"
+	buildDateLabel     string = "build_date"
+	serviceHealthName  string = "service_health"
+	// Adapt Ligato status code for now.
+	// TODO: Consolidate with that from the "Common Container Telemetry" proposal.
+	//serviceHealthHelp    string = "The health of the serviceLabel 0 = INIT, 1 = UP, 2 = DOWN, 3 = OUTAGE"
+	serviceHealthHelp    string = "The health of the serviceLabel 0 = INIT, 1 = OK, 2 = ERROR"
+	dependencyHealthName string = "service_dependency_health"
+	// Adapt Ligato status code for now.
+	// TODO: Consolidate with that from the "Common Container Telemetry" proposal.
+	//dependencyHealthHelp string = "The health of the dependencyLabel 0 = INIT, 1 = UP, 2 = DOWN, 3 = OUTAGE"
+	dependencyHealthHelp string = "The health of the dependencyLabel 0 = INIT, 1 = OK, 2 = ERROR"
+	serviceInfoName      string = "service_info"
+	serviceInfoHelp      string = "Build info for the service.  Value is always 1, build info is in the tags."
+)
+
+var (
+	httpPort string
+)
+
+// init is here only for parsing program arguments
+func init() {
+	flag.StringVar(&httpPort, "prometheus-http-port", rest.DefaultHTTPPort,
+		"Listening port for the Agent's Prometheus health/metrics port.")
+}
+
+// Plugin struct holds all plugin-related data.
+type Plugin struct {
```

Comment:
I would rename it to health/probe/PrometheusPlugin.

I think it is not too late to rename also existing health/probe/Plugin (I assume it is just used in rpcflavor) to health/probes/LivenessPlugin...

Reply:
ok with probe/PrometheusPlugin.